community/values/src/main/java/org/neo4j/values/storable/Values.java

```diff
@@ -84,6 +84,9 @@
     public static final ArrayValue EMPTY_DOUBLE_ARRAY = Values.doubleArray( new double[0] );
     public static final TextArray EMPTY_TEXT_ARRAY = Values.stringArray();
 
+    public static boolean nanoPrecisionConfigAltered;
```

Comment:
If we make this private and change its value in the `setNanoPrecision` method we will also be safe against the mistake of setting it to false anyway (like the current mistake in the shutdown() method in the lifecycle).